pkg/storegateway/bucket.go

```diff
@@ -1626,10 +1561,6 @@ func (s *BucketStore) LabelValues(ctx context.Context, req *storepb.LabelValuesR
 // Notice that when no matchers are provided, the list of matched postings is AllPostings,
 // so we could also intersect those with each label's postings being each one non-empty and leading to the same result.
 func blockLabelValues(ctx context.Context, b *bucketBlock, postingsStrategy postingsSelectionStrategy, maxSeriesPerBatch int, labelName string, matchers []*labels.Matcher, logger log.Logger, stats *safeQueryStats) ([]string, error) {
-	// This index reader shouldn't be used for ExpandedPostings, since it doesn't have the correct strategy.
-	labelValuesReader := b.loadedIndexReader(ctx, selectAllStrategy{}, stats)
-	defer runutil.CloseWithLogOnErr(b.logger, labelValuesReader, "close block index reader")
-
```

Comment:
this probably didn't belong here in the first place. We should be able to use `b.loadedIndexReader()` below instead of `b.indexReader()`